games-arcade/fishsupper/files/fishsupper-0.1.6-int16.patch

```diff
@@ -0,0 +1,51 @@
+Fix build with clang
+https://bugs.gentoo.org/739090
+
+--- a/src/SFont_gl.cpp
++++ b/src/SFont_gl.cpp
+@@ -172,9 +172,9 @@ void Mafferyew::SFont_gl::init(const std::string& image_file)
+     SDL_SetAlpha(image, 0, 0);
+     font_height = image->h - 1;
+     
+-    int x = 0;
+-    int begin = 0;
+-    int end = 0;
++    Sint16 x = 0;
++    Sint16 begin = 0;
++    Sint16 end = 0;
+     Uint32 pink = SDL_MapRGB(image->format, 255, 0, 255);
+     
+     while ( x < image->w )
+@@ -203,10 +203,10 @@ void Mafferyew::SFont_gl::init(const std::string& image_file)
+ // **************************************************
+ 
+ void Mafferyew::SFont_gl::create_font_char(SDL_Surface* image, 
+-                                        int begin_x, 
+-                                        int end_x)
++                                        Sint16 begin_x, 
++                                        Sint16 end_x)
+ {
+-    int font_char_w = end_x - begin_x;
++    Uint16 font_char_w = end_x - begin_x;
+     int conv_w = Useful_gl::power_of_two(font_char_w);
+     int conv_h = Useful_gl::power_of_two(font_height);    
+     SDL_Surface* converted;
+--- a/src/SFont_gl.h
++++ b/src/SFont_gl.h
+@@ -88,13 +88,13 @@ class SFont_gl
+         std::vector<Font_char> my_font_chars;
+         int spacing;
+         // Accessible via public interface - see 'text_height' above.
+-        int font_height;    
++        Uint16 font_height;    
+         
+         void init(const std::string& image_file);
+         
+         void create_font_char(SDL_Surface* image, 
+-                              int begin_x, 
+-                              int end_x);
++                              Sint16 begin_x, 
++                              Sint16 end_x);
```

Comment:
you're actually changing type signatures here. Any reason not to just add the `static_cast<>`s?

Reply:
Done.